File: JAVA/单例模式不同用户数据共享/java+servlet是数据共享/src/com/test/SingletonDemo.java
package com.test;

/**
 * Created by daiya on 2017/8/31.
 */
public class SingletonDemo {
    private int count;
    private static final SingletonDemo SINGLETON_DEMO=new SingletonDemo();
    private SingletonDemo(){
        this.count=0;
    }

    public static SingletonDemo getInstance(){
        return SINGLETON_DEMO;
    }

    public void incrCount(int incr){
        this.count+=incr;
    }

    public int getCount() {
        return count;
    }

    public void setCount(int count) {
        this.count = count;
    }
}

File: JAVA/单例模式不同用户数据共享/php+nginx的环境下不支持单例数据共享/singletondemo.php
<?php

namespace demo;

class SingletonDemo{
	private $count;
	private static $__instance;
	private function __construct(){
		$this->count = 0;
	}

	public static function getInstance(){
		if(self::$__instance == null || !isset(self::$__instance)){
			self::$__instance = new SingletonDemo();
		}

		return self::$__instance;
	}

	public function incrCount($incr){
		$this->count+=$incr;
	}

	public function getCount(){
		return $this->count;
	}
}

